### DISCUSSION TITLE
Dynamic Recipients causing slowdown

### DISCUSSION THREAD

**Discussion by chrisrowe** [solspace/craft-freeform#1094] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**

It seems when our form contains a dynamic recipients field the submission sends the email directly on save which is slowing down the response when the SMTP server is slow/unavalible. When I remove the dynamic recipients field the form posts instantly.

I assumed Freeform would create a queue job but it seems not, is there a setting or something I should look for to remedy this?

Cheers!

**Comment by kjmartens** [comment]
Hi @chrisrowe,

Can you confirm which version of Freeform you're using? And in your experiencing, is it just Dynamic Recipients slowing down the form submit, or any email notification feature? 🙂

**Comment by chrisrowe** [comment]
Hey, triggering notificaitons from the listing locks up the CMS until SMTP timeouts too. We're on 3.13.34 (latest on Craft 3) is this something you still support or do we need to push the client to upgrade to C4?

**Comment by kjmartens** [comment, marked as answer]
@chrisrowe,

We are going to make a queue option for handling email notifications in Freeform, but it will be added to Freeform 5 (minimum version Craft 4). We will continue to support Freeform 3, but typically just security and critical bug fixes.
